lib/pbench/server/api/resources/upload_api.py

```diff
@@ -128,35 +173,53 @@ def put(self, controller):
             )
             dataset.add()
         except DatasetDuplicate:
-            self.logger.info("Dataset already exists {}", filename)
+            self.logger.info(
+                "Dataset already exists, user = {}, ctrl = {}, file = {}",
+                username,
+                controller,
+                filename,
+            )
             response = jsonify(dict(message="Dataset already exists"))
-            response.status_code = 200
+            response.status_code = HTTPStatus.OK
             return response
         except Exception:
-            self.logger.exception("unable to create dataset for {}", filename)
+            self.logger.exception(
+                "unable to create dataset for user = {}, ctrl = {}, file = {}",
+                username,
+                controller,
+                filename,
+            )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
         if tar_full_path.is_file() or md5_full_path.is_file():
             self.logger.error(
-                "Dataset or corresponding md5 file already present on the disc, {}",
-                filename,
+                "Dataset, or corresponding md5 file, already present; tar {} ({}), md5 {} ({})",
+                tar_full_path,
+                "present" if tar_full_path.is_file() else "missing",
+                md5_full_path,
+                "present" if md5_full_path.is_file() else "missing",
             )
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
-        self.logger.info("Uploading file {} to {}", filename, dataset)
+        self.logger.info(
```

Comment:
Note that this is essentially the audit log. I could definitely see a unit test checking that this _particular_ INFO message is present except where a WARN or ERROR is expected ...

Reply:
I like the thinking, but for one log statement in this PR?

Reply:
Perhaps not for only one statement (although, once we have the mechanism, it'll be easy to use in other places which should have it!), but is there only _one_ in these changes?  I got the impression that there were several (but maybe that assumption was misplaced...).